src/core/impl.scala

```diff
@@ -118,24 +118,32 @@ trait SealedTraitDerivation:
 
   protected transparent inline def subtypesFromMirror[A, SubtypeTuple <: Tuple](
       m: Mirror.SumOf[A],
-      idx: Int = 0
+      idx: Int = 0 //no longer used, kept for bincompat
   ): List[SealedTrait.Subtype[Typeclass, A, _]] =
     inline erasedValue[SubtypeTuple] match
       case _: EmptyTuple =>
         Nil
       case _: (s *: tail) =>
-        new SealedTrait.Subtype(
-          typeInfo[s],
-          IArray.from(anns[s]),
-          IArray.from(inheritedAnns[s]),
-          IArray.from(paramTypeAnns[A]),
-          isObject[s],
-          idx,
-          CallByNeed(summonFrom {
-            case tc: Typeclass[`s`] => tc
-            case _                  => deriveSubtype(summonInline[Mirror.Of[s]])
-          }),
-          x => m.ordinal(x) == idx,
-          _.asInstanceOf[s & A]
-        ) :: subtypesFromMirror[A, tail](m, idx + 1)
+        val sub = summonFrom {
+          case mm: Mirror.SumOf[`s`] =>
```

Comment:
I don't quite understand what didn't work before? I guess if the subtype is itself a sum, it would get derived by the recursive call to `deriveSubtype`?

Reply:
yes, it would get derived, but the subtypes will not be in `subtypes` and there is no way to recover them

Reply:
hm, anyway for complex hierarchies which don't have to form a nice tree that's probably the best we can do anyway; thanks :)